### ISSUE TITLE
replaying a built-in returned a different result than expected

### ISSUE THREAD

**Issue by alfert** [parapluu/Concuerror#46]
I got this error message while running concurerror on an Elixir program: 

```
Error: On step 193, replaying a built-in returned a different result than expected:
  original: {event,<0.82.0>,
       {receive_event,{message,{code_server,{module,'Elixir.Observer.PID'}},
                               #Ref<0.0.1.98476>,0},
                      #Fun<code_server.3.111414683>,<0.82.0>,infinity,false},
       #Ref<0.0.1.99737>,
       [146,{file,"code_server.erl"}],
       [{message_received,#Ref<0.0.1.98476>}]}
  new     : {event,<0.82.0>,
       {receive_event,{message,{code_server,{module,'Elixir.Observer.PID'}},
                               #Ref<0.0.1.99179>,0},
                      #Fun<code_server.3.111414683>,<0.82.0>,infinity,false},
       #Ref<0.0.1.99737>,
       [146,{file,"code_server.erl"}],
       [{message_received,#Ref<0.0.1.99179>}]}

Please notify the developers, as this is a bug of Concuerror.

Done! (Exit status: error)
  Summary: 1635 errors, 1635/2008 interleavings explored
```
I run concuerror with the following script: 

```
#!/usr/bin/env bash 
#
# runs Concuerror 
#
set +x
# Path to Concuerror
CONC=../../erlang/Concuerror/concuerror

# Path to Elixir
ELIXIR=../elixir-1.0.2/lib/elixir/ebin
EXUNIT=../elixir-1.0.2/lib/ex_unit/ebin

# Path to our program under test
RX=_build/test/lib/reaxive/ebin

# Our Module
MOD=Elixir.Reaxive

# Our Test function
FUN=test

# Our Graph-Result
GRAPH=trace/concuerror.dot

###############################################
###############################################

$CONC --graph $GRAPH --show_races true --after_timeout 1000 \
	--depth_bound 200  --non_racing_system code_server --non_racing_system user \
	--pa $ELIXIR --pa $EXUNIT --pa $RX -m $MOD -t $FUN $*
```
The source code under test is here: https://github.com/alfert/reaxive/commit/e4bdad6e3bd36ef90ff3544ada68e975bc246091

**Comment by aronisstav** [comment]
Hi Klaus!

As it seems it will be difficult to replicate exactly what is going on there, could you please try a few things to minimize the problematic case? Namely:

* Is it the case that some traces are reaching the depth bound and hence are reported as erroneous?
* If it is, could you add `--ignore_error depth_bound` to your script and run it again to see if there are any other errors?
* Your script seems to be lacking the `--ignore_first_crash` flag, but you are using that too, right?
* Can you try to run the test with an `--interleaving_bound` of, say, 10 and `--show_races` and send the `concuerror_report.txt` file?

Thank you for the report and apologies for the delay in this reply!

Stavros

**Comment by alfert** [comment]
Hi Stavros, 

interesting results with your suggestions. At first, I had some problems to reproduce the situation. Apparently, the former command-line was wrong, sorry! Some experimentation resulted in this  command-line: 

```
$CONC --graph $GRAPH --show_races true --after_timeout 1000 \
	--non_racing_system code_server --non_racing_system user \
	--ignore_first_crash  \
	--depth_bound 200 \
	--strict_scheduling \
	--ignore_error depth_bound \
	--pa $ELIXIR --pa $EXUNIT --pa $RX -m $MOD -t $FUN $*
```
I played around with `--interleaving_bound`, finally reaching 1600, but the bug did not appear. However, without `--interleaving_bound` the bug appeared again. 

Since the `concuerror_report.txtt` file is rather large (4.3 MB), I'll send you a compressed  version to info@concuerror.com. 

Klaus.

**Comment by aronisstav** [comment]
Hi Klaus,

It seems that I never got your other mail. Can you send it again?

**Comment by alfert** [comment]
Hi Stavros,

Sorry for my slow response. The mail returned after a few days with an error. Is your mail server properly running or do you have another email address?

Thanks 
Klaus 

Von meinem iPhone gesendet

**Comment by aronisstav** [comment]
Hi Klaus,

You can use my [private](mailto:aronisstav@gmail.com) e-mail address instead. Could you also forward me the error message that you got, so that I can investigate the issue with my DNS provider?

Apologies for the trouble,
Stavros

**Comment by aronisstav** [comment]
No need to forward the error after all. The info@concuerror.com e-mail should be working now.

**Comment by alfert** [comment]
Done!

Von meinem iPhone gesendet

**Comment by aronisstav** [comment]
It should be the case that this issue is now resolved, after the merge of #84. I am therefore closing it, but feel free to reopen if it is not the case.
